internal/resources/kustomization/resource_kustomization.go

```diff
@@ -40,3 +63,376 @@ var kustomizationSchema = map[string]*schema.Schema{
+	config := m.(authctx.TanzuContext)
+
+	kustomizationName, ok := d.Get(nameKey).(string)
+	if !ok {
+		return diag.Errorf("unable to read kustomization name")
+	}
+
+	kustomizationNamespaceName, ok := d.Get(namespaceNameKey).(string)
+	if !ok {
+		return diag.Errorf("unable to read kustomization namespace name")
+	}
+
+	scopedFullnameData := scope.ConstructScope(d, kustomizationName, kustomizationNamespaceName)
+
+	if scopedFullnameData == nil {
+		return diag.Errorf("Unable to get Tanzu Mission Control kustomization entry; Scope full name is empty")
+	}
+
+	kustomizationDataFromServer, err := retrieveKustomizationDataFromServer(config, scopedFullnameData, d)
+	if err != nil {
+		if clienterrors.IsNotFoundError(err) {
+			_ = schema.RemoveFromState(d, m)
+			return
+		}
+
+		return diag.FromErr(err)
+	}
+
+	// always run
+	d.SetId(kustomizationDataFromServer.UID)
+
+	if err := d.Set(common.MetaKey, common.FlattenMeta(kustomizationDataFromServer.meta)); err != nil {
+		return diag.FromErr(err)
+	}
+
+	var (
+		flattenedSpec   []interface{}
+		flattenedStatus interface{}
+	)
+
+	switch scopedFullnameData.Scope {
+	case commonscope.ClusterScope:
+		flattenedSpec = spec.FlattenSpecForClusterScope(kustomizationDataFromServer.atomicSpec)
+		flattenedStatus = status.FlattenStatusForClusterScope(kustomizationDataFromServer.clusterScopeStatus)
+	case commonscope.ClusterGroupScope:
+		clusterGroupScopeSpec := &kustomizationclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupNamespaceFluxcdKustomizationSpec{
+			AtomicSpec: kustomizationDataFromServer.atomicSpec,
+		}
+		flattenedSpec = spec.FlattenSpecForClusterGroupScope(clusterGroupScopeSpec)
+		flattenedStatus = status.FlattenStatusForClusterGroupScope(kustomizationDataFromServer.clusterGroupScopeStatus)
+	}
+
+	if err := d.Set(spec.SpecKey, flattenedSpec); err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set(statusKey, flattenedStatus); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return diags
+}
+
+// nolint: gocognit
+func retrieveKustomizationDataFromServer(config authctx.TanzuContext, scopedFullnameData *scope.ScopedFullname, d *schema.ResourceData) (*dataFromServer, error) {
+	var kustomizationDataFromServer = &dataFromServer{}
+
+	// nolint: dupl
+	switch scopedFullnameData.Scope {
+	case commonscope.ClusterScope:
+		if scopedFullnameData.FullnameCluster != nil {
+			resp, err := config.TMCConnection.ClusterKustomizationResourceService.VmwareTanzuManageV1alpha1ClusterFluxcdKustomizationResourceServiceGet(scopedFullnameData.FullnameCluster)
+			if err != nil {
+				if clienterrors.IsNotFoundError(err) {
+					d.SetId("")
+					return kustomizationDataFromServer, err
+				}
+
+				return kustomizationDataFromServer, errors.Wrapf(err, "Unable to get Tanzu Mission Control cluster kustomization entry, name : %s", scopedFullnameData.FullnameCluster.Name)
+			}
+
+			scopedFullnameData.FullnameCluster = resp.Kustomization.FullName
```

Comment:
can these assignments be moved out, looks like this is duplicate in both cases

Reply:
It looks as if a duplicate due to the `resp` name for the variable being the same. The structure is different for both cluster and cluster group levels so moving the assignments out to make it generic is not possible.